packages/utils/src/api.ts

```diff
@@ -460,3 +460,10 @@ export function toFastProperties(toBecomeFast) {
   // tslint:disable-next-line
   eval(toBecomeFast)
 }
+
+export function upperFirst(str: string): string {
+  if (str) {
+    return str[0].toUpperCase() + str.substring(1)
```

Comment:
However there is an edge case where a single unicode character in JavaScript string would be represented by
a **surrogate** pair of characters.

```javascript
// pile of poo is just one character in unicode
// https://www.fileformat.info/info/unicode/char/1f4a9/index.htm
console.log("💩".length); // 2
```

In that case we will need to return 

```javascript
    return str.substring(0, 2).toUpperCase() + str.substring(2)
```
But I am not sure how to detect a surrogate pair

Reply:
Reference:
- https://dmitripavlutin.com/what-every-javascript-developer-should-know-about-unicode/#24-surrogate-pairs

This reference includes the range definition for possible surrogate pair values so I think a single range check for the first 2 characters could be used to detect this edge case.

Reply:
huh... you learn something every day.  I'll fix that up